### PR TITLE
google-protobuf: Updated definitions to version 3.7.1

### LINKED ISSUE #34825: [@types/google-protobuf] Missing definitions for ExtensionRangeOptions in descriptor_pb.d.ts
If you know how to fix the issue, make a pull request instead.

- [x] I tried using the `@types/google-protobuf` package and had problems.
- [x] I tried using the latest stable version of tsc. https://www.npmjs.com/package/typescript
- [x] I have a question that is inappropriate for [StackOverflow](https://stackoverflow.com/).  (Please ask any appropriate questions there).
- [x] [Mention](https://github.com/blog/821-mention-somebody-they-re-notified) the authors (see `Definitions by:` in `descriptor_pb.d.ts`) so they can respond.
  - Authors: @liujisi @MarcusLongmuir

-----------
Hi! 
I tried to use definitions from `@types/google-protobuf/google/protobuf/descriptor_pb.d.ts` and discovered that `ExtensionRangeOptions` and it usage were missed. 
Declaration of  `ExtensionRangeOptions`  could be found in [descriptor.proto](https://github.com/protocolbuffers/protobuf/blob/master/src/google/protobuf/descriptor.proto). Please, update `descriptor_pb.d.ts` according to it.

### PR DESCRIPTION
**Pull request by kamthamc**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/protocolbuffers/protobuf/compare/3.7.x...3.2.x
- [x] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

Closes #34825

### CODE REVIEW

**Comment by sandersn** [comment]
@MarcusLongmuir mind taking a look at these changes?

@kamthamc Can you add some tests for the new definitions?

**Review by sandersn — CHANGES_REQUESTED** [review]
Need some tests for new definitions.

**Comment by kamthamc** [comment]
@sandersn added tests for new definitions

**Review by sandersn — APPROVED** [review]

**Comment by sandersn** [comment]
Looks fine to me, but I don't know about google-protobuf, just Definitely Typed. @MarcusLongmuir can you give an opinion, bearing in mind this updates the definitions to 3.7? They were previously 3.2, I think.

File: types/google-protobuf/google/protobuf/struct_pb.d.ts [hunk 1]
```diff
@@ -1,12 +1,11 @@
+// package: google.protobuf
+// file: struct.proto
+
 import * as jspb from "../../index";
 
 export class Struct extends jspb.Message {
   getFieldsMap(): jspb.Map<string, Value>;
   clearFieldsMap(): void;
-
-  toJavaScript(): {[key: string]: JavaScriptValue};
```
Comment (ofpiyush) [inline types/google-protobuf/google/protobuf/struct_pb.d.ts]: Add this method back please

File: types/google-protobuf/google/protobuf/timestamp_pb.d.ts [hunk 2]
```diff
@@ -7,9 +10,6 @@ export class Timestamp extends jspb.Message {
   getNanos(): number;
   setNanos(value: number): void;
 
-  toDate(): Date;
```
Comment (ofpiyush) [inline types/google-protobuf/google/protobuf/timestamp_pb.d.ts]: Add these methods back please

File: types/google-protobuf/google/protobuf/struct_pb.d.ts [hunk 3]
```diff
@@ -94,10 +89,7 @@ export class ListValue extends jspb.Message {
   clearValuesList(): void;
   getValuesList(): Array<Value>;
   setValuesList(value: Array<Value>): void;
-  addValues(value?: Value, index?: number): void;
-
-  toJavaScript(): Array<JavaScriptValue>;
```
Comment (ofpiyush) [inline types/google-protobuf/google/protobuf/struct_pb.d.ts]: Add these methods back please

**Review by ofpiyush — COMMENTED** [review]
Please add the methods that are present in google-protobuf back

File: types/google-protobuf/google/protobuf/any_pb.d.ts [hunk 4]
```diff
@@ -9,10 +12,6 @@ export class Any extends jspb.Message {
   getValue_asB64(): string;
   setValue(value: Uint8Array | string): void;
 
-  getTypeName(): string;
```
[the remark this answers is not in the document]
Comment (lnieves) [reply types/google-protobuf/google/protobuf/any_pb.d.ts]: Yes, those methods are present even in google-protobuf 3.8.0.

Just wasted 2 hours figuring why a colleague could not compile her typescript code.

**Pull request merged**